From 95f78c0a102a858011dd605e851cdb0d2d51c293 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Mon, 18 Jul 2016 12:36:26 +0200
Subject: [PATCH 054/827] dmaengine: xilinx: Fix compilation error

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Remove has_sg checking which was removed by:
"dmaengine: xilinx: zynqmp_dma: Sync driver with mainline"
(sha1: 4cd70c71159ab9cc3555217bf2c1fba12cd23261)

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 6c95d00b074163af1b0641309e95564ffb8d0ada)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/dma/xilinx/zynqmp_dma.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/dma/xilinx/zynqmp_dma.c b/drivers/dma/xilinx/zynqmp_dma.c
index 8bd45d0..c5eff87 100644
--- a/drivers/dma/xilinx/zynqmp_dma.c
+++ b/drivers/dma/xilinx/zynqmp_dma.c
@@ -661,9 +661,6 @@ static void zynqmp_dma_free_chan_resources(struct dma_chan *dchan)
 	zynqmp_dma_free_descriptors(chan);
 	spin_unlock_bh(&chan->lock);
 
-	if (!chan->has_sg)
-		return;
-
 	dma_free_coherent(chan->dev,
 		(2 * ZYNQMP_DMA_DESC_SIZE(chan) * ZYNQMP_DMA_NUM_DESCS),
 		chan->desc_pool_v, chan->desc_pool_p);
-- 
2.9.3

